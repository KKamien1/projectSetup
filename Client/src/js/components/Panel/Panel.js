import React from "react";
import { FormattedMessage } from "react-intl";
import "./Panel.scss";

export default function Panel({ max, min, dots, onChange }) {
  return (
    <div className="panel">
      <input
        type="range"
        name="dots"
        min={min}
        max={max}
        value={dots}
        onChange={onChange}
      />
      <div>
        <FormattedMessage id="Panel.Dots" defaultMessage="Dots" />: {dots}
      </div>
    </div>
  );
}
