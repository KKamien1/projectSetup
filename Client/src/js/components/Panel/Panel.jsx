import React from 'react';
import PropTypes from 'prop-types';

export default function Panel({
  max, min, dots, onChange,
}) {
  return (
    <div className="panel">
      <input
        type="range"
        name="dots"
        min={min}
        max={max}
        value={dots}
        onChange={onChange}
      />
    </div>
  );
}

Panel.propTypes = {
  max: PropTypes.number,
  min: PropTypes.number,
  dots: PropTypes.number,
  onChange: PropTypes.func,
};
Panel.defaultProps = {
  max: 30,
  min: 0,
  dots: 5,
  onChange: () => {},
};
