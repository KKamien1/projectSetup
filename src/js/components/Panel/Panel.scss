.panel {
  position: absolute;
  bottom: 40px;
  left: 20px;
  z-index: 12;
}
