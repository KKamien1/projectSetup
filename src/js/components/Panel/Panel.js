import React, { useContext } from "react";
import "./Panel.scss";

export default function Panel({ max, min, dots, onChange }) {
  const stopAnimation = e => {
    e.preventDefault();
    console.log("stop");
    //cancelAnimationFrame(rAF.current);
  };

  return (
    <div className="panel">
      <input
        type="range"
        name="dots"
        min={min}
        max={max}
        value={dots}
        onChange={onChange}
      />
      <div>Value: {dots}</div>
      <div>
        <button onClick={stopAnimation}>Stop</button>
      </div>
    </div>
  );
} 