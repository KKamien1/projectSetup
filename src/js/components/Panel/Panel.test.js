import React from "react";
import { shallow } from "enzyme";

import Panel from "./Panel";
import { JestEnvironment } from "@jest/environment";

// import Context from "../../App";
// const { dispatch } = useContext(Context);

function setup(props = {}) {
  const defaultProps = {
    dots: 12
  };
  const context = {
    dispatch: jest.fn()
  };
  const wrapper = shallow(<Panel {...defaultProps} {...props} />, { context });
  return wrapper;
}

describe("Panel Test Suite!", () => {
  //let dispatch;
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
    console.log(wrapper.debug());
    //dispatch = jest.fn();
  });

  it("Should have an input of range type", () => {
    expect(wrapper.find("input[type='range']").exists()).toBe(true);
  });
  it("Should change a dots props", () => {
    wrapper.find("input").simulate("change", { target: { value: 25 } });
    expect(wrapper.instance().props.dots).toBe(25);
  });
});
