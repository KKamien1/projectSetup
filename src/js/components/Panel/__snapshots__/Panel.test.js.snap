// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Testing should render 1`] = `ShallowWrapper {}`;

exports[`Panel Testing should render correctly with props 1`] = `
<Panel
  dots={14}
/>
`;
